singularity-eos/eos/eos.hpp

```diff
@@ -129,7 +129,19 @@ class Gruneisen : public EosBase<Gruneisen> {
             const Real b, const Real rho0, const Real T0, const Real P0, const Real Cv,
             const Real rho_max)
       : _C0(C0), _s1(s1), _s2(s2), _s3(s3), _G0(G0), _b(b), _rho0(rho0), _T0(T0), _P0(P0),
-        _Cv(Cv), _rho_max(rho_max) {}
+        _Cv(Cv), _rho_max(rho_max) {
+    // Warn user when provided rho_max is greater than the computed rho_max
+#ifndef NDEBUG
+    const Real computed_rho_max = ComputeRhoMax(s1, s2, s3, rho0);
+    if (rho_max > RHOMAX_SAFETY * computed_rho_max) {
```

Comment:
Maybe worth also checking if `rho_max` is a NaN as `>` will return false in that case.

Reply:
There are no shortage of checks we could make.  Ensuring that `rho_max > rho0` seems promising.  But, there is a difference from a check that will ensure sane input (like this) or one that will ensure we don't sample regions that are mathematically unsound (`rho_max < computed_rho_max`).

Reply:
I would propose that we tackle the issue of sanitizing inputs in a separate MR since that applies to other EOS as well. I don't think the behavior of the Gruneisen EOS will be particularly worse than any other EOS for NaN input parameters given that we can just expect undefined behavior across the board.

I'll make an issue.

Reply:
https://github.com/lanl/singularity-eos/issues/131